app/src/main/java/org/mozilla/rocket/privately/browse/BrowserFragment.kt

```diff
@@ -425,56 +443,57 @@ class BrowserFragment : LocaleAwareFragment(),
+            chromeViewModel.onTabCountChanged(sessionManager.size)
         }
 
-        override fun onShowFileChooser(
-            es: TabViewEngineSession,
-            filePathCallback: ValueCallback<Array<Uri>>?,
-            fileChooserParams: WebChromeClient.FileChooserParams?
-        ): Boolean {
-            // do nothing, exist for interface compatibility only.
-            return false
-        }
-
-        var callback: FullscreenCallback? = null
-        var session: Session? = null
-
-        override fun onSessionAdded(session: Session, arguments: Bundle?) {
-        }
+        override fun onSessionSelected(session: Session) {
+            lastSession?.unregister(sessionObserver)
+            session.register(sessionObserver)
+            lastSession = session
 
-        override fun onProgress(session: Session, progress: Int) {
-            fragment.progressView.progress = progress
+            chromeViewModel.run {
+                onFocusedUrlChanged(session.url)
+                onFocusedTitleChanged(session.title)
+                onNavigationStateChanged(session.canGoBack, session.canGoForward)
+            }
         }
+    }
 
-        override fun onTitleChanged(session: Session, title: String?) {
-            fragment.chromeViewModel.onFocusedTitleChanged(title)
-            session.let {
-                if (fragment.displayUrlView.text.toString() != it.url) {
-                    fragment.displayUrlView.text = it.url
+    private val sessionObserver = object : Session.Observer {
+        override fun onDownload(session: Session, download: mozilla.components.browser.session.Download): Boolean {
+            activity.let {
+                if (it == null || !it.lifecycle.currentState.isAtLeast(Lifecycle.State.STARTED)) {
+                    return false
                 }
             }
-        }
 
-        override fun onLongPress(session: Session, hitTarget: HitTarget) {
-            fragment.activity?.let {
-                WebContextMenu.show(true,
-                        it,
-                        PrivateDownloadCallback(fragment, session.url),
-                        hitTarget)
-            }
+            val d = Download(
+                    download.url,
+                    download.fileName,
+                    download.userAgent,
+                    "",
+                    download.contentType,
+                    download.contentLength ?: 0,
+                    false
+            )
+            permissionHandler.tryAction(
+                    fragment,
+                    Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                    ACTION_DOWNLOAD,
+                    d
+            )
+            return true
         }
 
-        override fun onEnterFullScreen(callback: FullscreenCallback, view: View?) {
-            with(fragment) {
+        override fun onFullScreenChanged(session: Session, enabled: Boolean) {
+            if (enabled) {
```

Comment:
Every time I try to enter fullscreen mode vis JS or using Youtube(html5?), the app crashes with below error
```

2020-04-08 00:16:11.549 7067-7067/org.mozilla.rocket.coverage:private_mode A/chromium: [FATAL:jni_android.cc(249)] Please include Java exception stack in crash report
2020-04-08 00:16:11.549 7067-7067/org.mozilla.rocket.coverage:private_mode A/libc: Fatal signal 5 (SIGTRAP), code 128 in tid 7067 (ge:private_mode), pid 7067 (ge:private_mode)
```
Again this may not related to the code chage here. But still need to be aware of

Reply:
fixed